src/editor/Store/ConfigurationStoreConnector.re

```diff
@@ -31,7 +31,20 @@ let start = () => {
       switch (configPath) {
       | Ok(configPathAsString) =>
         switch (ConfigurationParser.ofFile(configPathAsString)) {
-        | Ok(v) => dispatch(Actions.ConfigurationSet(v))
+        | Ok(v) =>
+          dispatch(Actions.ConfigurationSet(v));
+
+          switch (cliOptions) {
+          | Some(cliOptions) =>
+            let zenModeSingleFile =
+              Configuration.getValue(c => c.zenModeSingleFile, v);
+
+            if (zenModeSingleFile && List.length(cliOptions.filesToOpen) == 1) {
+              dispatch(Actions.ToggleZenMode);
```

Comment:
I'm wondering if we want to add a `bool` to ToggleZenMode to be explicit that we enable it (or make it 2 actions EnableZenMode/DisableZenMode). Because now this really is a toggle, and we're assuming that we are the first to execute this action.
Making it explicit will make it more clear, and less prone to strange behaviour (because of the reload)

Reply:
Good idea, @bryphe should I change this in this PR?

Reply:
Maybe we should open an issue about this discussing it first?

Reply:
I like the idea of switching from `ToggleZenMode` to `EnableZenMode`/`DisableZenMode! Agree with @tcoopman that it would be a bit safer - instead of assuming we're the first ones to call toggle, just calling `EnableZenMode` means this logic will always work regardless of if it was called previously.

It would be a little clearer in the command palette too if we could show the right one - showing "Enable zen mode" and "Disable zen mode" conditionally is clearer than just "Toggle zen mode". 

I'm ok tracking it another issue, though 👍

Reply:
Opened an [issue](https://github.com/onivim/oni2/issues/634) to track it.